scripts/ci/test-downgrade.sh

```diff
@@ -0,0 +1,109 @@
+#!/usr/bin/env bash
+set -exo pipefail
+
+declare -a DOWNGRADE_TARGETS=('2.0')
+
+declare -r TEST_ORG=testorg
+declare -r TEST_TOKEN=supersecretfaketoken
+declare -r INIT_PING_ATTEMPTS=600
+
+function download_older_binary () {
+    local -r target_version=$1 dl_dir=$2
+
+    local dl_url
+    local dl_sha
+    case ${target_version} in
+        2.0)
+            dl_url=https://dl.influxdata.com/influxdb/releases/influxdb2-2.0.9-linux-amd64.tar.gz
+            dl_sha=64b9cfea1b5ca07479a16332056f9e7f806ad71df9c6cc0ec70c4333372b9d26
+            ;;
+        *)
+            >&2 echo Error: Unknown downgrade target "'$target_version'"
+            exit 1
+            ;;
+    esac
+
+    local -r archive="influxd-${target_version}.tar.gz"
+    curl -sL -o "${dl_dir}/${archive}" "$dl_url"
+    echo "${dl_sha}  ${dl_dir}/${archive}" | sha256sum --check --
+    tar xzf "${dl_dir}/${archive}" -C "$dl_dir" --strip-components=1
+    rm "${dl_dir}/${archive}"
+    mv "${dl_dir}/influxd" "${dl_dir}/influxd-${target_version}"
+}
+
+function test_downgrade_target () {
+    local -r influxd_path=$1 target_version=$2 tmp=$3
+
+    download_older_binary "$target_version" "$tmp"
+
+    local -r bolt_path="${tmp}/influxd-${target_version}.bolt"
+    local -r sqlite_path="${tmp}/influxd-${target_version}.sqlite"
+
+    cp "${tmp}/influxd.bolt" "$bolt_path"
+    cp "${tmp}/influxd.sqlite" "$sqlite_path"
+
+    "$influxd_path" downgrade \
+        --bolt-path "$bolt_path" \
+        --sqlite-path "$sqlite_path" \
+        "$target_version"
+
+    INFLUXD_BOLT_PATH="$bolt_path" INFLUXD_SQLITE_PATH="$sqlite_path" INFLUXD_ENGINE_PATH="${tmp}/engine" \
+        "${tmp}/influxd-${target_version}" &
+    local -r influxd_pid="$!"
+
+    wait_for_influxd "$influxd_pid"
```

Comment:
After the old version starts up, can we do some kind simple check to make verify that it has started up with the downgraded metadata files? Maybe just confirming that an HTTP request requiring authentication works for the same credentials that were used in the `setup_influxd` command. That would be a very basic sanity check that the old version is using the same bolt file. Doesn't seem like there would be anything analogous for the sqlite file since the downgrade to `2.0` will totally clear it out though.

Reply:
Added a check that `/api/v2/me` works